### PR TITLE
Graphics: Tone mapping is now done in a separate class

### PR DESCRIPTION
**Pull request by samcake**
- splitting the tone mapping job in its own class and file away from DeferredLightingEffect

- Removing uneeded stuff from DeferredLightingEffect

- Re arranging the job order to do all the cpu only jobs upfront and the gpu dependant jobs last

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 9ddc3c27bf7e35ad4f963f949741415ce682bacd available at: https://gist.github.com/1691e61d64e7cb5e9530<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6582.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6582.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 9faec326da023116577a45ef0a3792bb2894c875 available at: https://gist.github.com/1691e61d64e7cb5e9530<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6582.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6582.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: ede375a8895ede27a7848e0063f6779d390c9e14 available at: https://gist.github.com/1691e61d64e7cb5e9530<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6582.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6582.dmg

File: libraries/render-utils/src/RenderDeferredTask.cpp [hunk 1]
```diff
@@ -44,67 +44,85 @@ void RenderDeferred::run(const SceneContextPointer& sceneContext, const RenderCo
...
         FetchItems(
-            [] (const RenderContextPointer& context, int count) {
-                context->_numFeedOpaqueItems = count; 
-            }
+        [](const RenderContextPointer& context, int count) {
+        context->_numFeedOpaqueItems = count;
+    }
         )
-    )));
+        )));
```
Comment (AndrewMeadows) [inline libraries/render-utils/src/RenderDeferredTask.cpp]: The indentation looks messed around line 56.

**Comment by hifi-gustavo** [comment]
Build console output for commit: c74bbda663028c8ec621c899396899891931e5f4 available at: https://gist.github.com/1691e61d64e7cb5e9530<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/High+Fidelity-PR-6582-win64.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6582.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: a50b14dfae6d6c055ebd6965e2b784e2f9772e18 available at: https://gist.github.com/1691e61d64e7cb5e9530<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/High+Fidelity-PR-6582-win64.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6582.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: e70ab472aff62972eb6f22011c25af54339ef669 available at: https://gist.github.com/1691e61d64e7cb5e9530<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/High+Fidelity-PR-6582-win64.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6582.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 2bbd5d86b8f72e092c68aed2a0c7fa8bd88ecb2d available at: https://gist.github.com/1691e61d64e7cb5e9530<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/High+Fidelity-PR-6582-win64.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6582.dmg

**Pull request merged**